src/ui/public/vis/agg_configs.js

```diff
@@ -126,6 +135,15 @@ export default function AggConfigsFactory(Private) {
       }
     });
 
+    function removeParentAggs(obj) {
+      for(const prop in obj) {
+        if (prop === 'parentAggs') delete obj[prop];
+        else if (typeof obj[prop] === 'object') removeParentAggs(obj[prop]);
+      }
+    }
+
+    removeParentAggs(dslTopLvl);
```

Comment:
This is easier to read than the other example. Here to, consider moving outside the scope so we're not instantiating this function over and over. Again, up to you, I don't feel strongly about it.